docs/ProfilingModes.md

```diff
@@ -0,0 +1,136 @@
+
+Sampling interval can be adjusted with `--alloc` option.
+For example, `--alloc 500k` will take one sample after 500 KB of allocated
+space on average. Prior to JDK 11, intervals less than TLAB size will not take effect.
+
+### Installing Debug Symbols
+
+Prior to JDK 11, the allocation profiler required HotSpot debug symbols.
+Some OpenJDK distributions (Amazon Corretto, Liberica JDK, Azul Zulu)
+already have them embedded in `libjvm.so`, other OpenJDK builds typically
+provide debug symbols in a separate package. For example, to install
+OpenJDK debug symbols on Debian / Ubuntu, run:
+```
+# apt install openjdk-17-dbg
+```
+(replace `17` with the desired version of JDK).
+
+On CentOS, RHEL and some other RPM-based distributions, this could be done with
+[debuginfo-install](http://man7.org/linux/man-pages/man1/debuginfo-install.1.html) utility:
+```
+# debuginfo-install java-1.8.0-openjdk
+```
+
+On Gentoo the `icedtea` OpenJDK package can be built with the per-package setting
+`FEATURES="nostrip"` to retain symbols.
+
+The `gdb` tool can be used to verify if debug symbols are properly installed for the `libjvm` library.
+For example, on Linux:
+```
+$ gdb $JAVA_HOME/lib/server/libjvm.so -ex 'info address UseG1GC'
+```
+This command's output will either contain `Symbol "UseG1GC" is at 0xxxxx`
+or `No symbol "UseG1GC" in current context`.
+
+## Wall-clock profiling
+
+`-e wall` option tells async-profiler to sample all threads equally every given
+period of time regardless of thread status: Running, Sleeping or Blocked.
+For instance, this can be helpful when profiling application start-up time.
+
+Wall-clock profiler is most useful in per-thread mode: `-t`.
+
+Example: `asprof -e wall -t -i 5ms -f result.html 8983`
+
+## Java method profiling
+
+`-e ClassName.methodName` option instruments the given Java method
+in order to record all invocations of this method with the stack traces.
+
+Example: `-e java.util.Properties.getProperty` will profile all places
+where `getProperty` method is called from.
+
+Only non-native Java methods are supported. To profile a native method,
+use hardware breakpoint event instead, e.g. `-e Java_java_lang_Throwable_fillInStackTrace`
+
+**Be aware** that if you attach async-profiler at runtime, the first instrumentation
+of a non-native Java method may cause the [deoptimization](https://github.com/openjdk/jdk/blob/bf2e9ee9d321ed289466b2410f12ad10504d01a2/src/hotspot/share/prims/jvmtiRedefineClasses.cpp#L4092-L4096)
+of all compiled methods. The subsequent instrumentation flushes only the _dependent code_.
+
+The massive CodeCache flush doesn't occur if attaching async-profiler as an agent.
+
+Here are some useful native methods that you may want to profile:
+* ```G1CollectedHeap::humongous_obj_allocate``` - trace _humongous allocations_ of the G1 GC,
+* ```JVM_StartThread``` - trace creation of new Java threads,
+* ```Java_java_lang_ClassLoader_defineClass1``` - trace class loading.
```

Comment:
These should be a in a separate subsection: Native function profiling.

Reply:
Corrected. Will reflect in next commit